src/search/search_dictionary.go

```diff
@@ -12,18 +14,43 @@ import (
 // tokenizing the content, and populating the WordOccurrences map with word frequencies.
 // It takes a pointer to an IndexListEntry as input and returns an error if any issues occur
 // during content fetching or processing.
-func CreateSearchDictionary(ile *pb.IndexListEntry) error {
 
+func CreateSearchDictionary(ile *pb.IndexListEntry) error {
 	content, err := fetch.FetchFromSource(ile.SourceType, ile.URI)
 	if err != nil {
-		return fmt.Errorf("failed to read source file: %w", err)
+		return fmt.Errorf("failed to open source file: %w", err)
+	}
+
+	if len(content) == 0 {
+		return nil
 	}
 
-	// Create and populate the word_occurrences map
+	fileContent := string(content)
+
 	ile.WordOccurrences = make(map[string]int32)
-	tokens := strings.Fields(strings.ToLower(string(content)))
-	for _, token := range tokens {
-		ile.WordOccurrences[token]++
+	ile.StemmedWordOccurrences = make(map[string]int32)
+
+	parser := tokenizer.New()
+	parser.AllowKeywordUnderscore()
+
+	// parse file content
```

Comment:
remove obvious comments like this

Reply:
i didnt think this was that obvious, but i changed it to be //parse & tokenize file content